parsing/lexer.mll

```diff
@@ -627,6 +662,10 @@ and string = parse
   | '\\' 'x' ['0'-'9' 'a'-'f' 'A'-'F'] ['0'-'9' 'a'-'f' 'A'-'F']
       { store_escaped_char lexbuf (char_for_hexadecimal_code lexbuf 2);
          string lexbuf }
+  | '\\' 'u' '{'
+        hex_digit hex_digit? hex_digit? hex_digit? hex_digit? hex_digit? '}'
```

Comment:
Will do the `hex_digit+`. 

Regarding `_` I'd rather not, this would diverge from the standard `\u{h}`notation (which I would like to stress I did invent out of thin air).